### DISCUSSION TITLE
Hide RAG References.

### DISCUSSION THREAD

**Discussion by Snify89** [open-webui/open-webui#5899] [General]
**Is your feature request related to a problem? Please describe.**
No. This is just an idea I'd like to have implemented.

**Describe the solution you'd like**
It would be great to have an option to "hide" the references coming from the RAG. If you use doc search, you see the references and the clear text from the document. There might be cases, where this is unwanted. E.g. I have "sensitive" contents that the LLM should filter out, yet I can see the entire documents.

**Describe alternatives you've considered**
I have a workaround by feeding the system prompt with the most important content needed. This needs a higher context size.

**Additional context**
I hope you understand what I mean.
Thank you for reading and considering my idea/suggestion. Keep up your good work. :)

**Comment by ticpu** [comment]
Not sure this is really what you want, if the AI has access to the content, there's always someone who will find a way to make the AI rewrite that content verbatim.  This works until it doesn't, removing the sensitive information might be the only solution there.

**Reply by Snify89** [reply]
Thank you for discussing this.
The intention in my specific case is a creative one. I'd like to share some kind of diary but I'd also like to hide the real references. Maybe I can find a workaround via funtion/pipes or even modifying/changing certain HTML objects.

**Comment by demon-ru** [comment]
I am also interested in this functionality.
My case is this: data from many clients is used for RAG knowledge. And the response never contains any sensitive data from the RAG. But in the RAG chunks they are there and visible.
Yes, I understand that it is necessary to anonymize data when adding it to RAG. But this is not being done yet.
Therefore, a temporary solution would be not to show chunks from the knowledge base.
